### PR TITLE
Feat/history

### PR DESCRIPTION
**Pull request by abhij1607**
This PR consist of

- history page
- delete single video from history
- delete all single video from history

### CODE REVIEW

File: src/pages/History/HistoryVideoCard/HistoryVideoCard.jsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import { requestDeleteVideoInHistory } from "../../../utils/server-request";
+import { useAuth } from "../../../context/auth-context";
+
+const HistoryVideoCard = ({ video }) => {
+  const {
+    userState: { userToken },
+    userDispatch,
+  } = useAuth();
+
+  const headers = { headers: { authorization: userToken } };
```
Comment (Hamdrive) [inline src/pages/History/HistoryVideoCard/HistoryVideoCard.jsx:12]: Does not have to be initialized here, can instead be directly set in the API call itself

**Review by shivampandey0 — APPROVED** [review]
Good implementation for adding video to History and managing history with remove and removeAll

Suggestion: could add some message on History page for no videos to make it more user friendly

**Review by Hamdrive — COMMENTED** [review]
Good work done implementing history feature and deletion of videos
Have suggested better handling for API calls

File: src/pages/History/HistoryVideoCard/HistoryVideoCard.jsx (hunk 1 above)
Comment (abhij1607) [reply src/pages/History/HistoryVideoCard/HistoryVideoCard.jsx:12]: did this to make code more clean and readable

**Pull request merged**